src/app/modules/datagrid/datagrid-controller.js

```diff
@@ -1,135 +1,252 @@
 (function () { // to wrap use strict
 	'use strict';
 
+	//This variable allows both the onExit function of the stateProvider and the controller to access the resizeGrid function
+	var resizeGrid;
+
+	var tagList = [];
+
 	// Model declaration for the data grid view (url #/datagrid)
-	angular.module('BloomLibraryApp.datagrid', ['ui.router', 'restangular', 'ngGrid'])//, 'BloomLibraryApp.detail'])
+	angular.module('BloomLibraryApp.datagrid', ['ui.router', 'restangular', 'ui.grid', 'ui.grid.pagination', 'ui.grid.resizeColumns', 'ui.grid.edit', 'ui.grid.cellNav', 'ui.grid.autoResize', 'ngTagsInput'])//, 'BloomLibraryApp.detail'])
 	.config(['$stateProvider', function config($stateProvider) {
 
 		$stateProvider.state('datagrid', {
+			parent: 'requireLoginResolution',
 			url: "/datagrid",
 			templateUrl: 'modules/datagrid/datagrid.tpl.html',
-			controller: 'DataGridCtrl'
+			controller: 'DataGridCtrl',
+			title: 'Book Library',
+			onExit: function() {
+				window.removeEventListener("resize", resizeGrid);
+			}
 		});
 	} ]);
 
+	angular.module('BloomLibraryApp.datagrid')
+	.service('autoTags', ['tagService', '$q', function(tagService, $q) {
```

Comment:
I finally figured out this is for auto-complete. I think it would be worth actually naming it autoCompleteTags...it's not used enough to make a long name a burden.